hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -647,13 +650,37 @@ public String reconfigurePropertyImpl(String property, String newVal)
       }
       break;
     }
+    case DFS_DATANODE_MAX_RECEIVER_THREADS_KEY:
+      return reconfDataXceiverParameters(property, newVal);
     default:
       break;
     }
     throw new ReconfigurationException(
         property, newVal, getConf().get(property));
   }
 
+  private String reconfDataXceiverParameters(String property, String newVal)
+      throws ReconfigurationException {
+    String result;
+    try {
+      LOG.info("Reconfiguring {} to {}", property, newVal);
+      if (property.equals(DFS_DATANODE_MAX_RECEIVER_THREADS_KEY)) {
```

Comment:
The if-statement seems redundant since it is already checked by `case DFS_DATANODE_MAX_RECEIVER_THREADS_KEY `.

Reply:
Thanks @tasanuma for your review. I agree with your suggestion. There is only one parameter at present. I will fix it.